tonic/models/vic/grid_params.py

```diff
@@ -661,64 +1246,178 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
-            new = np.zeros(shape) + FILLVALUE_F
-            new[:-1, :, :] = out_dicts['veg_dict'][var]
-            new[-1, :, :] = 0
-            out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
+        #   double fcan(veg_class, month, nj, ni) ;
+        #   double albedo(veg_class, month, nj, ni) ;
+        fill_val = FILLVALUE_F
+        varnames = ['root_depth', 'root_fract']
+        if vegparam_lai and lai_src == 'FROM_VEGPARAM':
+            varnames.append('LAI')
+        if vegparam_fcan and fcan_src == 'FROM_VEGPARAM':
+            varnames.append('fcanopy')
+        if vegparam_albedo and alb_src == 'FROM_VEGPARAM':
+            varnames.append('albedo')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
+            new = np.full(shape, fill_val)
+            if extra_class:
+                new[:-1, :, :] = out_dicts['veg_dict'][var]
+                new[-1, :, :] = bare_vegparam[var]
+            else:
+                new[:, :, :] = out_dicts['veg_dict'][var]
+            out_dicts['veg_dict'][var] = np.ma.masked_values(new, fill_val)
+
+        if blowing_snow:
+            for var in ['sigma_slope', 'lag_one', 'fetch']:
+                shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
+                new = np.full(shape, fill_val)
+                if extra_class:
+                    new[:-1, :, :] = out_dicts['veg_dict'][var]
+                    new[-1, :, :] = bare_vegparam[var]
+                else:
+                    new[:, :, :] = out_dicts['veg_dict'][var]
+                out_dicts['veg_dict'][var] \
+                    = np.ma.masked_values(new, fill_val)
 
         # Distribute the veglib variables
         # 1st - the 1d vars
         #   double lib_overstory(veg_class) ;  --> (veg_class, nj, ni)
-        for var in ['overstory', 'rarc', 'rmin', 'wind_h', 'RGL', 'rad_atten',
-                    'rad_atten', 'wind_atten', 'trunk_ratio', 'snow_albedo']:
+        varnames = ['overstory', 'rarc', 'rmin', 'wind_h', 'RGL', 'rad_atten',
+                    'rad_atten', 'wind_atten', 'trunk_ratio']
+        if veglib_photo:
+            varnames.append('Ctype')
+            varnames.append('MaxCarboxRate')
+            varnames.append('MaxE_or_CO2Spec')
+            varnames.append('LUE')
+            varnames.append('Nscale')
+            varnames.append('Wnpp_inhib')
+            varnames.append('NPPfactor_sat')
+        for var in varnames:
             lib_var = 'lib_{0}'.format(var)
-            new = np.zeros((nveg_clases, ysize, xsize)) + FILLVALUE_F
-            new[:-1, yi, xi] = veglib_dict[lib_var][:, np.newaxis]
-            new[-1, yi, xi] = 0
+            if var in ['Ctype', 'Nscale']:
+                fill_val = FILLVALUE_I
+                dtype_str = 'np.int'
+            else:
+                fill_val = FILLVALUE_F
+                dtype_str = 'np.float'
```

Comment:
Ah, OK, I guess they are like pre-compiler constants in C/Fortran, rather than strings.  Makes sense.